com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/JarSignatureVerifier.java

```diff
@@ -0,0 +1,175 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.JarURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Splitter;
+import com.sap.cloud.lm.sl.cf.core.util.FileUtils;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.SLException;
+
+@Component
+public class JarSignatureVerifier {
+
+    private static final String META_INF = "META-INF";
+    private static final String CN = "CN";
+    private static final int BUFFER_SIZE = 8 * 1024;
+
+    public void verify(List<X509Certificate> targetCertificates, String certificateCN, URL jarURL) {
+        try {
+            JarFile jarFile = openJarFile(jarURL);
+            List<JarEntry> jarEntries = getJarEntries(jarFile);
+            verifyJarEntries(targetCertificates, jarEntries, certificateCN);
```

Comment:
The order of the arguments of this method is not intuitive. `targetCertificates` and `certificateCN` are both used as verification parameters (i.e. how things **should** be), while `jarEntries` are the things that must be verified. Change the signature so that the `jarEntries` come first.